crates/holochain_persistence_api/src/eav/query.rs

```diff
@@ -154,32 +154,39 @@ impl<'a, A: Attribute> EaviQuery<'a, A> {
 }
 
 /// Represents a filter type which takes in a function to match on
-pub struct EavFilter<'a, T: 'a + Eq>(Box<dyn Fn(T) -> bool + 'a>);
+// pub struct EavFilter<'a, T: 'a + Eq>(Box<dyn Fn(T) -> bool + 'a>);
+pub enum EavFilter<'a, T: 'a + Eq> {
+    Exact(T),
```

Comment:
This is required so we can explicitly know if a query is an exact match on the entity and optimize

Reply:
Hmm, I'm ok with this but makes me wonder how the api is going to ultimately extend over time.

Reply:
This is cool, when I wrote these types I was hoping to see them extended just like this when the time to optimize came.